### ISSUE TITLE
Can't read an HDFS file

### ISSUE THREAD

**Issue by srajeev** [databricks/spark-avro#180]
Hello -

I am trying to read an avro HDFS file with this:

```
val sqlContext = new SQLContext(sparkCtx)
 val df = sqlContext.read.format("com.databricks.spark.avro").load("./hdfs://localhost:8020/testavro/00a099c9-0446-4e74-98e9-c1e364019872.avro")
```

My file can be seen in local HDFS here: http://screencast.com/t/SD5j5iSnM4yQ

But it is giving this exception. 

```
java.lang.IllegalArgumentException: java.net.URISyntaxException: Relative path in absolute URI: file:C:/Users/srajeev/IdeaProjects/IMS.DT.DataFileUtilsAPI/spark-warehouse
    at org.apache.hadoop.fs.Path.initialize(Path.java:206) ~[hadoop-common-2.2.0.jar:na]
    at org.apache.hadoop.fs.Path.<init>(Path.java:172) ~[hadoop-common-2.2.0.jar:na]
    at org.apache.spark.sql.catalyst.catalog.SessionCatalog.makeQualifiedPath(SessionCatalog.scala:114) ~[spark-catalyst_2.11-2.0.0.jar:2.0.0]
```

I also tried like this:

```
val df = sparkSession.read.avro("hdfs://localhost:8020/testavro/00a099c9-0446-4e74-98e9-c1e364019872.avro")
```

Thanks
Shyla

**Comment by koertkuipers** [comment]
what is the error if you run:
`val df = sparkSession.read.avro("hdfs://localhost:8020/testavro/00a099c9-0446-4e74-98e9-c1e364019872.avro")`

**Comment by JoshRosen** [comment]
It sounds like you might be hitting https://issues.apache.org/jira/browse/SPARK-15899, which is a but in Spark 2.0.0 that is supposed to be fixed in Spark 2.0.1. There should also be workarounds listed on that JIRA ticket.

**Comment by vehementer-e** [comment]
may be 8020 is not an appropriate port
try 9000 instead  - hdfs://<host>:9000/path_in_hdfs
